assets/javascripts/discourse/components/babble-upload.js.es6

```diff
@@ -0,0 +1,58 @@
+import { default as computed, on, observes } from 'ember-addons/ember-computed-decorators';
+import { uploadTranslate } from 'discourse/controllers/upload-selector';
+
+export default Ember.Component.extend({
+  tagName: "div",
+  classNameBindings: [':babble-upload', ':babble-modal', 'visible::hidden'],
+  local: true,
+  imageUrl: null,
+  showMore: false,
+  remote: Ember.computed.not("local"),
+
+  @on('didInsertElement')
+  @observes('local')
+  selectedChanged() {
+    Ember.run.next(() => {
+      // *HACK* to select the proper radio button
+      const value = this.get('local') ? 'local' : 'remote';
+      $('input:radio[name="upload"]').val([value]);
+      $('.inputs input:first').focus();
+    });
+  },
+
+  @computed()
+  title() {
+    return uploadTranslate("title");
+  },
+
+  close: function() {
+    this.set('visible', false)
+  },
+
+  actions: {
+    upload: function() {
+      const $composer = this.$().parent().siblings('.babble-post-composer')
+      if (this.get('local')) {
+        $composer.fileupload('add', { fileInput: this.$('#filename-input') });
+      } else {
+        const imageUrl = this.get('imageUrl') || '';
+        const imageLink = imageUrl.substr(imageUrl.lastIndexOf('/') + 1)
+        this.sendAction('addText', `![${imageLink}](${imageUrl})`)
+      }
+      this.close()
```

Comment:
This closes the modal whether there's a file put into the upload or not, I think. Would be nice to have an error message saying you should upload something.